internal/templates/assisted-installer/template.go

```diff
@@ -106,8 +106,19 @@ kind: InfraEnv
 metadata:
   annotations:
     siteconfig.open-cluster-management.io/sync-wave: "1"
-  name: "{{ .Spec.ClusterName }}"
+{{ if .SpecialVars.CurrentNode.NodeRef }}
+  name: "{{ .SpecialVars.CurrentNode.NodeRef.Name }}"
+  namespace: "{{ .SpecialVars.CurrentNode.NodeRef.Namespace }}"
+{{ else }}
+  name: "{{ .SpecialVars.CurrentNode.HostName }}"
   namespace: "{{ .Spec.ClusterName }}"
+{{ end }}
+  labels:
+{{ if .SpecialVars.CurrentNode.NodeRef }}
+    baremetalhost.metal3.io/namespace: "{{ .SpecialVars.CurrentNode.NodeRef.Namespace }}"
```

Comment:
What is this? I don't think this is required.

Reply:
Removed!